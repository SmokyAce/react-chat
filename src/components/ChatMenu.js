import React from 'react';
import PropTypes from 'prop-types';
import IconButton from 'material-ui/IconButton';
import Menu, { MenuItem } from 'material-ui/Menu';
import MoreVertIcon from 'material-ui-icons/MoreVert';

class ChatMenu extends React.Component {
  static propTypes = {
    user: PropTypes.shape({
      isCreator: PropTypes.bool.isRequired,
    }).isRequired,
    isConnected: PropTypes.bool.isRequired,
    actions: PropTypes.shape({
      deleteChat: PropTypes.func.isRequired,
      leaveChat: PropTypes.func.isRequired,
    }).isRequired,
  };

  state = {
    anchorEl: null,
  };

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { actions, user, isConnected } = this.props;

    const action = {
      name: user.isCreator ? 'Delete' : 'Leave',
      onClick: user.isCreator ? actions.deleteChat : actions.leaveChat,
    };

    return (
      <React.Fragment>
        <IconButton
          aria-label="More"
          aria-owns={anchorEl ? 'chat-menu' : null}
          aria-haspopup="true"
          onClick={this.handleClick}
          color="inherit"
          disabled={!isConnected}
        >
          <MoreVertIcon />
        </IconButton>
        <Menu
          id="long-menu"
          anchorEl={this.state.anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem
            onClick={() => {
              action.onClick();
              this.handleClose();
            }}
          >
            {action.name}
          </MenuItem>
        </Menu>
      </React.Fragment>
    );
  }
}

export default ChatMenu;
