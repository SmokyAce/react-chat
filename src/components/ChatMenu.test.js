/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMenu from './ChatMenu';

const mockProps = {
  user: {
    isCreator: false,
  },
  isConnected: false,
  actions: {
    deleteChat: jest.fn(),
    leaveChat: jest.fn(),
  },
};

describe('<ChatMenu />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMenu {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatMenu {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
