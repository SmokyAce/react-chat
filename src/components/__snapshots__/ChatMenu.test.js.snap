// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMenu /> renders correctly 1`] = `
<button
  aria-haspopup="true"
  aria-label="More"
  aria-owns={null}
  className="MuiButtonBase-root-29 MuiButtonBase-disabled-30 MuiIconButton-root-23 MuiIconButton-colorInherit-24 MuiIconButton-disabled-27"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiIconButton-label-28"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-32"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </span>
</button>
`;
